src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast.hidden === false) {
+      beast.health -= 50;
+    }
```

Comment:
<img width="641" alt="image" src="https://user-images.githubusercontent.com/102293862/215693610-e7390043-f1a9-4b58-88fd-dac3bdccc19f.png">
carnivore shouldn't be able to bite another carnivore